pkg/tests/testprovider/component.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016-2021, Pulumi Corporation.
```

Comment:
Nit, I think these are copied from pu/pu, since the provider needs to be available at a relative path. Not sure if we need to update the copyright dates here and in the other files.

```suggestion
// Copyright 2024, Pulumi Corporation.
```